game/src/main/scala/io/aigar/game/EntityContainer.scala

```diff
@@ -2,21 +2,35 @@ package io.aigar.game
 
 import com.github.jpbetz.subspace.Vector2
 
-
 trait EntityContainer {
-  def respawn(grid: Grid, players: List[Player], respawnRetryAttempts: Int): Option[Vector2] = {
-    var newPosition = grid.randomPosition
+  def shouldRespawn: Boolean
 
+  def getRespawnPosition(grid: Grid, players: List[Player], respawnRetryAttempts: Int): Option[Vector2] = {
     1 to respawnRetryAttempts foreach { _ =>
+      var collides = false
+      val newPosition = grid.randomPosition
       for (player <- players) {
         for (cell <- player.cells) {
-          if (cell.contains(newPosition))
-            newPosition = grid.randomPosition
-          else
-            return Some(newPosition)
+          if (cell.contains(newPosition)) collides = true
         }
       }
+      if (!collides) return Some(newPosition)
     }
     None
   }
+
+  def detectCollisions(entities: List[Entity], players: List[Player]): List[Entity] ={
+    var entitiesReturn = entities
+    for(entity <- entities){
```

Comment:
I think this could be done with a [filter](https://twitter.github.io/scala_school/collections.html#filter), but it's good enough